anta/inventory/__init__.py

```diff
@@ -494,8 +492,7 @@ def create_device_session(self, host_ip: str) -> bool:
         Returns:
             bool: True if update succeed, False if not
         """
-        logger.debug(
-            f'Searching for device {host_ip} in {[str(dev.host) for dev in self._inventory]}')
+        logger.debug(f'Searching for device {host_ip} in {[str(dev.host) for dev in self._inventory]}')
```

Comment:
So in these next couple of lines we compute again and again some iterators

I would suggest to replace it like this

```suggestion
        logger.debug(f'Searching for device {host_ip} in inventory')
```

and then:

the next few lines to be

```
        device = [dev for dev in self._inventory if str(dev.host) == str(host_ip)][:1] or None
        if device is None:
            return False
        logger.debug(f'Search result is: {device}')
        if device.is_online and not device.established and self._is_ip_exist(host_ip):
            logger.debug(f'Trying to connect to device {str(device.host)}')
            device = self._build_device_session(device=device, timeout=self.timeout)
            # pylint: disable=W0104
            [device if dev.host == device.host else dev for dev in self._inventory]
            return True
```

Though I am not quite sure what this line purpose is:
```
[device if dev.host == device.host else dev for dev in self._inventory]
```
Looks like some list that is never used (not assign to anything) and the method returms True or False

Reply:
- Regarding first change in the log with inventory, Idea was to give a method to manually check consistence of the search in debug mode. So it is the reason why IP addresses of the inventory are listed. If you think it is useless, we can remove it. (@ksator as we had the exact same discussion internally)
- Other changes are fixed now.